### PR TITLE
fix(time) Catch functions without timezone parameter

### PR DESCRIPTION
**Pull request by evanh**
If one of the time granularity functions was called in a condition without a
timezone parameter, we were not correctly translating the value to a datetime,
causing Clickhouse errors when the value was not formatted correctly. Add an
extra case to catch those expressions and translate them as well.

### CODE REVIEW

**Review by alxu-sentry — COMMENTED** [review]
can't comment on snuba logic, but it looks like exactly what we need for sentry. thanks!

File: snuba/query/processors/timeseries_processor.py [hunk 1]
```diff
@@ -57,11 +57,23 @@ def __init__(
                                 [
                                     String("toStartOfHour"),
                                     String("toStartOfMinute"),
+                                    String("toStartOfDay"),
```
Comment (fpacifici) [inline snuba/query/processors/timeseries_processor.py:60]: Please add it here as well. 
https://github.com/getsentry/snuba/blob/master/snuba/query/processors/timeseries_processor.py#L208

File: snuba/query/processors/timeseries_processor.py [hunk 2]
```diff
@@ -57,11 +57,23 @@ def __init__(
...
                                     String("toDate"),
                                 ]
                             ),
                             (column_match, LiteralMatch(Any(str))),
```
Comment (fpacifici) [inline snuba/query/processors/timeseries_processor.py:64]: Do we need to match these two cases (this one and the one you added) explicitly? Isn't the one below + optional parameters enough ?

Comment (evanh) [reply snuba/query/processors/timeseries_processor.py:64]: You're right, just the one is fine.

File: snuba/query/processors/timeseries_processor.py [hunk 3]
```diff
@@ -205,7 +207,14 @@ def extract_granularity_from_query(query: Query, column: str) -> Optional[int]:
...
+            Or(
+                [
+                    String("toStartOfHour"),
+                    String("toStartOfMinute"),
+                    String("toStartOfDay"),
+                    String("toDate"),
+                ]
+            ),
         ),
         (column_match, LiteralMatch(Any(str))),
```
Comment (fpacifici) [inline snuba/query/processors/timeseries_processor.py:219]: I think you need to turn this one as well into 
```
(column_match,),
with_optionals=True,
```

**Review by fpacifici — APPROVED** [review]

**Pull request merged**